crates/evm/src/tests/fork_tests.rs

```diff
@@ -0,0 +1,807 @@
+            CallMessage {
+                txs: vec![deploy_message],
+            },
+            &context,
+            &mut working_set,
+        )
+        .unwrap();
+    }
+    evm.end_soft_confirmation_hook(&soft_confirmation_info, &mut working_set);
+    evm.finalize_hook(&[99u8; 32].into(), &mut working_set.accessory_state());
+
+    l2_height += 1;
+
+    // Send money to transient storage contract
+    evm.begin_soft_confirmation_hook(&soft_confirmation_info, &mut working_set);
+    {
+        let context = C::new(sender_address, l2_height, SovSpecId::Genesis, l1_fee_rate);
+        let call_tx = call_mcopy(contract_addr, &dev_signer, 1);
+
+        evm.call(
+            CallMessage { txs: vec![call_tx] },
+            &context,
+            &mut working_set,
+        )
+        .unwrap();
+    }
+    evm.end_soft_confirmation_hook(&soft_confirmation_info, &mut working_set);
+    evm.finalize_hook(&[99u8; 32].into(), &mut working_set.accessory_state());
+
+    l2_height += 1;
+
+    let receipts: Vec<_> = evm
+        .receipts
+        .iter(&mut working_set.accessory_state())
+        .collect();
+
+    // Last tx should have failed because cancun is not activated
+    assert!(!receipts.last().unwrap().receipt.success);
+
+    let soft_confirmation_info = HookSoftConfirmationInfo {
+        l2_height,
+        da_slot_hash: [5u8; 32],
+        da_slot_height: 1,
+        da_slot_txs_commitment: [42u8; 32],
+        pre_state_root: [10u8; 32].to_vec(),
+        current_spec: SovSpecId::Fork1,
+        pub_key: vec![],
+        deposit_data: vec![],
+        l1_fee_rate,
+        timestamp: 0,
+    };
+
+    // Send money to transient storage contract
+    evm.begin_soft_confirmation_hook(&soft_confirmation_info, &mut working_set);
+    {
+        let context = C::new(sender_address, l2_height, SovSpecId::Fork1, l1_fee_rate);
+        let call_tx = call_mcopy(contract_addr, &dev_signer, 2);
+
+        evm.call(
+            CallMessage { txs: vec![call_tx] },
+            &context,
+            &mut working_set,
+        )
+        .unwrap();
+    }
+    evm.end_soft_confirmation_hook(&soft_confirmation_info, &mut working_set);
+    evm.finalize_hook(&[99u8; 32].into(), &mut working_set.accessory_state());
+
+    l2_height += 1;
+
+    let receipts: Vec<_> = evm
+        .receipts
+        .iter(&mut working_set.accessory_state())
+        .collect();
+
+    let txs: Vec<_> = evm
+        .transactions
+        .iter(&mut working_set.accessory_state())
+        .collect();
+
+    // Last tx should have failed because cancun is not activated
+    assert!(receipts.last().unwrap().receipt.success);
+    let db_account = DbAccount::new(contract_addr);
+    let storage_value = db_account
+        .storage
+        .get(&U256::ZERO, &mut working_set)
+        .unwrap();
+    assert_eq!(storage_value, U256::from(80));
+}
+
+// tests first part of https://eips.ethereum.org/EIPS/eip-6780
+#[test]
+fn test_self_destruct_restriction() {
```

Comment:
since this tests starts exactly the same as `self_destruct_test` i think you can just extend that test with what you put here.

keep the creation tx selfdestruct here though, but put a comment referencing where the other part of the cancun upgrade for selfdestruct was implemented